startup/00-base.py

```diff
@@ -113,112 +117,31 @@ def post_document(name, doc):
-            # to avoid circular reference preventing collection.
-            # NOTE: This still doesn't guarantee call on delete or gc.collect()!
-            #       Explicitly call flush() if immediate write to disk required.
-            def finalize(zfile, cache, dump):
-                zfile.update((k, dump(v)) for k, v in cache.items())
-
-            import weakref
-            self._finalizer = weakref.finalize(
-                self, finalize, self._file, self._cache, PersistentDict._dump)
-
-        @property
-        def directory(self):
-            return self._directory
-
-        def __setitem__(self, key, value):
-            self._cache[key] = value
-            super().__setitem__(key, value)
-
-        def __getitem__(self, key):
-            return self._cache[key]
-
-        def __delitem__(self, key):
-            del self._cache[key]
-            super().__delitem__(key)
-
-        def __repr__(self):
-            return f"<{self.__class__.__name__} {dict(self)!r}>"
-
-        @staticmethod
-        def _dump(obj):
-            "Encode as msgpack using numpy-aware encoder."
-            # See https://github.com/msgpack/msgpack-python#string-and-binary-type
-            # for more on use_bin_type.
-            return msgpack.packb(
-                obj,
-                default=msgpack_numpy.encode,
-                use_bin_type=True)
-
-        @staticmethod
-        def _load(file):
-            return msgpack.unpackb(
-                file,
-                object_hook=msgpack_numpy.decode,
-                raw=False)
-
-        def flush(self):
-            """Force a write of the current state to disk"""
-            for k, v in self.items():
-                super().__setitem__(k, v)
-
-        def reload(self):
-            """Force a reload from disk, overwriting current cache"""
-            self._cache = dict(super().items())
-
-
-# using appdirs line for xspress3 development on xf05id2-ws1
-# do not commit this
-# runengine_metadata_dir = appdirs.user_data_dir(appname="bluesky") / Path("runengine-metadata")
-# runengine_metadata_dir = Path('/nsls2/xf05id1/shared/config/runengine-metadata-new')
-runengine_metadata_dir = Path('/nsls2/data/srx/legacy/xf05id1/shared/config/runengine-metadata-new')
-
-from nslsii.md_dict import RunEngineRedisDict
-
-RE.md = PersistentDict(runengine_metadata_dir)
-
-# RE.md = RunEngineRedisDict()
+RE.md = RedisJSONDict(redis.Redis("info.srx.nsls2.bnl.gov"), prefix="bsui")
 
 # Optional: set any metadata that rarely changes.
 RE.md["beamline_id"] = "SRX"
 RE.md["md_version"] = "1.1"
 
+
+class SRXPrompt(Prompts):
+    def in_prompt_tokens(self, cli=None):
+        return [
+            (
+                Token.Prompt,
+                f"☁️  BlueSky@SRX | Proposal #{RE.md.get('proposal', {}).get('proposal_num', 'N/A')} [",
+            ),
+            (Token.PromptNum, str(self.shell.execution_count)),
+            (Token.Prompt, "]: "),
+        ]
+
+
+ip.prompts = SRXPrompt(ip)
+""
```

Comment:
```suggestion

```

Reply:
Not a huge deal, but I found this strange indentation behavior when you hit return. Perhaps limited to my terminal.

![image](https://github.com/NSLS-II-SRX/profile_collection/assets/25692496/69d86168-3427-46dc-85bf-fa18c9e85477)

Reply:
We can remove the ☁️ symbol, it was more like a joke.